wire-ios-data-model/Source/ManagedObjectContext/Migration/CoreDataMigrationActionFactory.swift

```diff
@@ -20,6 +20,34 @@ import Foundation
 
 enum CoreDataMigrationActionFactory {
 
+    static func createPreMigrationAction<Version: CoreDataMigrationVersion>(for destinationVersion: Version) -> CoreDataMigrationAction? {
+
+        if let version = destinationVersion as? CoreDataMessagingMigrationVersion {
+            return createPreMigrationAction(for: version)
+        }
+
+        if let version = destinationVersion as? CoreDataEventsMigrationVersion {
+            return createPreMigrationAction(for: version)
+        }
```

Comment:
We should probably refactor this in a way that doesn't rely on casting, but I think it's fine for now.

Reply:
yeah, I couldn't figure it out